### PR TITLE
Switch `NonZero` alias direction.

### PR DESCRIPTION
**Pull request by reitermarkus**
Step 4 mentioned in https://github.com/rust-lang/rust/pull/100428#pullrequestreview-1767139731.

Depends on https://github.com/rust-lang/rust/pull/120160.

r? @dtolnay

### CODE REVIEW

File: src/tools/clippy/clippy_lints/src/transmute/transmute_int_to_non_zero.rs [hunk 1]
```diff
@@ -16,40 +16,55 @@ pub(super) fn check<'tcx>(
...
+    let Some(nonzero_alias) = coercable_types
+        .iter()
+        .find_map(|(nonzero_alias, t)| {
+            if *t == int_type && *t == from_ty {
+                Some(nonzero_alias)
+            } else {
+                None
+            }
+        })
+        else { return false; };
```
Comment (kadiwa4) [inline src/tools/clippy/clippy_lints/src/transmute/transmute_int_to_non_zero.rs:55]:
```suggestion
    let Some((nonzero_alias, _)) = coercable_types
        .iter()
        .find(|&(_, t)| t == int_type && t == from_ty)
    else {
        return false;
    };
```
It's probably better if it passes rustfmt, even when editing the clippy subtree

Comment (reitermarkus) [reply src/tools/clippy/clippy_lints/src/transmute/transmute_int_to_non_zero.rs:55]: Is there a way to make `./x fmt` do this, or do I have to run `cargo fmt` in the subdirectory?

Comment (kadiwa4) [reply src/tools/clippy/clippy_lints/src/transmute/transmute_int_to_non_zero.rs]: Probably the latter, but I'm not sure.

**Review by dtolnay — APPROVED** [review]
Thank you!

**Comment by fmease** [comment]
I think this might have caused this failure in a rollup: https://github.com/rust-lang/rust/pull/120290#issuecomment-1907358039.
@bors r-

**Comment by reitermarkus** [comment]
@fmease, updated the `debuginfo` tests and `libcore.natvis` file.

**Comment by reitermarkus** [comment]
@matthiaskrgr, fixed the missing spaces in the test.

**Review by dtolnay — APPROVED** [review]

**Comment by dtolnay** [comment]
Feel free to use `r=dtolnay` if further test fixes are discovered.

**Review by dtolnay — CHANGES_REQUESTED** [review]
- Needs a rebase over #116167.

**Pull request merged**